### DISCUSSION TITLE
Most efficient way to resize multiple times

### DISCUSSION THREAD

**Discussion by SoftCircuits** [dlemstra/Magick.NET#1718] [Help] [answered]
If I have a `MagickImage` instance, and from that I want to create two images with different sizes, what is the most efficient way to do this?

I figure I could clone the original image and then resize that. But that seems like it would use a lot of memory.

I also see the `Clone()` method takes a `MagickGeometry` parameter. Is resizing the image during cloning every bit as efficient as a reguler `Resize()`. Is there a recommendation?

**Comment by dlemstra** [comment]
Your question is similar to what was discussed in this issue: https://github.com/dlemstra/Magick.NET/issues/1577. With the release of version 14.0.0 changes were made to the source generator and I think I can take a look at these changes again.

**Comment by SoftCircuits** [comment]
@dlemstra Thanks, but by "I can take a look at these changes again", does that mean the feature is not implemented?

It looked like your `Clone()` had a version that would resize. But I don't know about the implementation or which version would be more efficient. Was hoping for some insight.

**Reply by dlemstra** [reply, marked as answer]
The issue is still open so yes it has not been implemented. The `Clone` overload with a `MagickGeometry` should be called `CloneArea` and only clones part of the image. It doesn't resize it. For now you will need to wait for that other issue to be implemented.

**Comment by SoftCircuits** [comment]
@dlemstra Thanks for the clarification. So, at this point, I am cloning the image and then resizing the clone for each size I need.

    IMagickImage image = Image.Clone();
    image.Resize(maxX, maxY);
    image.Format = MagickFormat.Jpeg;
    image.Quality = 80;
    using MemoryStream memoryStream = new();
    image.Write(memoryStream);
    return memoryStream.ToArray();

My main concern here is that, with larger images, this might not be the most efficient use of memory. If there is a more efficient way, please let me know.

**Reply by dlemstra** [reply]
Right now there is no interface/api to do this more efficient. Make sure you watch the issue that I mentioned because I will report back there when that api has been implemented.
